packages/firebase/src/handlers/check-store-events.ts

```diff
@@ -128,7 +128,17 @@ export default async () => {
       if (isOrdersOnly) {
         return;
       }
-      if (lastNonOrdersTimestamp) {
+      if (listenedEventName === 'carts-delayed') {
+        // defines the limits for getting events with predefined delay
+        const delayCart = process.env.DELAY_CART_MIN
+          ? parseInt(process.env.DELAY_CART_MIN, 10)
+          : (1000 * 60 * 5);
+
+        params['timestamp>'] = new Date((lastNonOrdersTimestamp || lastRunTimestamp) - delayCart)
+          .toISOString();
+
+        params['timestamp<'] = new Date(timestamp - delayCart).toISOString();
+      } else if (lastNonOrdersTimestamp) {
         params['timestamp>'] = new Date(lastNonOrdersTimestamp).toISOString();
```

Comment:
```suggestion
      if (lastNonOrdersTimestamp) {
        if (listenedEventName.endsWith('delayed')) {
          const delayMs = process.env.API_EVENTS_DELAYED_MS
            ? parseInt(process.env.API_EVENTS_DELAYED_MS, 10)
            : (1000 * 60 * 5);
          params['timestamp>'] = new Date(lastNonOrdersTimestamp - delayMs).toISOString();
          params['timestamp<'] = new Date(timestamp - delayMs).toISOString();
        } else {
          params['timestamp>'] = new Date(lastNonOrdersTimestamp).toISOString();
        }
```

O ponto importante aí é o `process.env.API_EVENTS_DELAYED_MS`, nome de variável global é API do software, tem que ser descritivo, não-tão-verboso, e principalmente consistente. Pra variável de ambiente pense sempre em um prefixo, nesse caso usei `API_EVENTS_*`, quando precisarmos de outra variável de ambiente de relacionada a eventos da API usaremos ele novamente.

Fora isso acho que a gente pode manter tudo dentro do `if (lastNonOrdersTimestamp) {` só pra simplificar porque na verdade `lastNonOrdersTimestamp` na primeira execução do primeiro deploy...